app/views/users/show.html.erb

```diff
@@ -151,10 +151,13 @@
         <%= image_tag("three-dots.svg", class: "dropdown-icon", alt: "Toggle dropdown menu") %>
       </button>
       <div id="user-profile-dropdownmenu" class="profile-dropdownmenu">
-        <button id="user-profile-dropdownmenu-block-button" data-profile-user-id="<%= @user.id %>" class="block-button" type="button">Block @<%= @user.username %></button>
         <% if user_signed_in? %>
-          <%= javascript_pack_tag "profileDropdown", defer: true %>
-        <% end %>
+          <button id="user-profile-dropdownmenu-block-button" data-profile-user-id="<%= @user.id %>" type="button">Block @<%= @user.username %></button>
+          <button id="user-profile-dropdownmenu-abuse-button" data-profile-user-id="<%= @user.id %>" type="button" style="display: none;">
+            <%= Reaction.cached_any_reactions_for?(@user, current_user, "vomit") ? "Remove Abuse Flag" : "Flag as Abusive" %>
```

Comment:
As discussed, this will need to be changed to client-side logic similar to how the block logic is created.

Probably need to create an endpoint to ask whether someone has already reacted on a user.

We could also scrap this part of it for now and make it only available on the `/mod` page. I think that would be at least a fine interim decision, and maybe just the best thing overall as mentioned in a past call, maybe better to just have one place to do this.